### ISSUE TITLE
Red glitch before lock screen shows up

### ISSUE THREAD

**Issue by hengtseChou** [YaLTeR/niri#808]
I am trying to pick a locking utility for the niri setup. However, with swaylock and hyprlock, they glitch a red screen very briefly before proceeding to display the lock screen normally. I believe it is very similar to issue [#341](https://github.com/YaLTeR/niri/issues/341)?

I have tried out niri 0.1.9 on arch extra and 0.10.1 on arch extra-testing, and they both showing same glitch.

Thank you for this amazing project!

### System Information

* niri version: 0.10.1

* Distro: Arch 6.11.8-arch1-2

* GPU: Intel Iris Xe Graphics

* CPU:Intel i5-13500H

**Comment by YaLTeR** [comment]
Hey, it's not a glitch, it's niri's locked session background. It shows up because niri doesn't wait for the lock screen to present a surface before drawing the locked session. This is tracked as part of #119.

With hyprlock specifically, there's an extra problem that it animates from transparency, so niri's red locked session background shows up during the animation. Sway seems to use a black background under lock surfaces so you don't notice it there. I'm not sure if the protocol mandates using a black background in this case.

**Comment by lpnh** [comment]
My approach to this is to use a script calling [do-screen-transition](https://github.com/YaLTeR/niri/wiki/Configuration%3A-Key-Bindings#do-screen-transition) before the lock screen command:

```fish
#!/usr/bin/env fish

niri msg action do-screen-transition --delay-ms 300

swaylock -f
```

To bind a script in your niri config, using the `spawn` action:

```
binds {
    Mod+T { spawn "~/scripts/do-something.sh"; }
}
```

**Comment by hengtseChou** [comment]
@lpnh Appreciate the workaround!

**Comment by hengtseChou** [comment]
@YaLTeR Thanks for the clarification 😌

**Comment by YaLTeR** [comment]
Ah yeah, the screen transition workaround, thanks!

**Comment by salman-farooq-sh** [comment]
Can we hardcode a black background instead of the dark red one until https://github.com/YaLTeR/niri/issues/119 is done? @YaLTeR

**Comment by YaLTeR** [comment]
Yeah, here: https://github.com/YaLTeR/niri/blob/8665003269d1fbe4efe3c477a71400392930cac9/src/niri.rs#L155

**Comment by ypurpl** [comment]
> Fixed in [4f16be9](https://github.com/YaLTeR/niri/commit/4f16be9e4d3b13925f5fb5368f4e24cba50227c6) (hyprlock will still show red bc it fades from transparency, so the red background shows through)

sorry if i misunderstood, but wouldn't it show black instead of red?

**Comment by YaLTeR** [comment]
The lockscreen protocol doesn't explicitly say what background the compositor should use when there's a semitransparent lock surface, so niri keeps using red.
